src/chainstate/stacks/db/queryable_logging.rs

```diff
@@ -0,0 +1,42 @@
+//! The methods in this file implement a "queryable" form for logging the outcomes of transactions.
+//!
+//! Each processed transaction should be logged if it succeeds, logged if it fails with an error,
+//! and should be logged each time it is skipped.
+//!
+//! This way, when debugging a miner, we can run a standard query, and see what happened to the
+//! transaction.
+//!
+use chainstate::stacks::Error;
+/// Logs a queryable message for the case where `txid` has succeeded.
+use chainstate::stacks::StacksTransaction;
+
+/// Returns a prefix for a "queryable logging" statement. This is the "queryable" part, which is to
+/// say that the prefix will be the same no matter what the outcome is, so that a fixed regex will
+/// match this part for any transaction.
+///
+/// Format looks like this:
+/// "Transaction outcome 491493d01fc8c0f21f0891ad7c378bfb9c7b83142524f3320ea609e6b5cffa90: "
+fn create_transaction_key(tx: &StacksTransaction) -> String {
```

Comment:
If we're going through the effort of building a queryable logging system for transactions, you might as well use the existing structured logging logic to ensure that downstream consumers receive a semantic representation of the transaction status (i.e. something machine-readable).